lib/src/test/java/com/dynatrace/file/util/FilePollerTest.java

```diff
@@ -27,32 +83,95 @@ void filePollerConstructorThrowsOnPassingAFolder() throws IOException {
 
     assertThrows(
         IllegalArgumentException.class,
-        () -> {
-          new FilePoller(tempDir.toString());
-        });
+        () -> FilePollerFactory.getPollBased(tempDir.toString(), Duration.ofMillis(50)));
+
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> FilePollerFactory.getWatchServiceBased(tempDir.toString()));
+  }
+
+  @Test
+  void filePollerUpdatesOnChangePollBased() throws IOException {
+    filePollerUpdatesOnChange(
+        FilePollerFactory.getPollBased(tempFiles.get(0).toString(), Duration.ofMillis(50)));
   }
 
   @Test
-  void filePollerUpdatesOnChange() throws IOException, InterruptedException {
-    final Path tempFile = Files.createTempFile("tempfile", ".txt");
+  void filePollerUpdatesOnChangeWatchServiceBased() throws IOException {
+    if (isMacOs) {
+      System.out.println("Running on macOS, skipping this test...");
```

Comment:
Maybe:

```suggestion
      System.out.println("macOS does not support WatchService based pollers, skipping this test...");
```
I think this might make it a bit easier to understand why the test is skipped :)